src/ophyd_async/testing/_one_of_everything.py

```diff
@@ -57,48 +54,64 @@ def float_array_signal(
         ],
         dtype=dtype,
     )
-    return soft_signal_rw(Array1D[dtype], value, name)
+
+
+@dataclass
+class EverythingSignal:
+    name: str
+    dtype: type[SignalDatatype]
+    initial_value: Any = None
+
+
+def get_every_signal_data():
+    # list containing necessary info to construct a signal of each type for multiple
+    # transports e.g. soft/epics/tango
+    return [
+        EverythingSignal("int", int, 1),
+        EverythingSignal("float", float, 1.234),
+        EverythingSignal("str", str, "test_string"),
+        EverythingSignal("bool", bool, True),
+        EverythingSignal("enum", ExampleEnum, ExampleEnum.B),
+        EverythingSignal("int8a", Array1D[np.int8], int_array_value(np.int8)),
+        EverythingSignal("uint8a", Array1D[np.uint8], int_array_value(np.uint8)),
+        EverythingSignal("int16a", Array1D[np.int16], int_array_value(np.int16)),
+        EverythingSignal("uint16a", Array1D[np.uint16], int_array_value(np.uint16)),
+        EverythingSignal("int32a", Array1D[np.int32], int_array_value(np.int32)),
+        EverythingSignal("uint32a", Array1D[np.uint32], int_array_value(np.uint32)),
+        EverythingSignal("int64a", Array1D[np.int64], int_array_value(np.int64)),
+        EverythingSignal("uint64a", Array1D[np.uint64], int_array_value(np.uint64)),
+        EverythingSignal(
+            "float32a", Array1D[np.float32], float_array_value(np.float32)
+        ),
+        EverythingSignal(
+            "float64a", Array1D[np.float64], float_array_value(np.float64)
+        ),
+        EverythingSignal("stra", Sequence[str], ["one", "two", "three"]),
+        EverythingSignal(
+            "enuma", Sequence[ExampleEnum], [ExampleEnum.A, ExampleEnum.C]
+        ),
+        EverythingSignal(
+            "table",
+            ExampleTable,
+            ExampleTable(
+                bool=np.array([False, False, True, True], np.bool_),
+                int=np.array([1, 8, -9, 32], np.int32),
+                float=np.array([1.8, 8.2, -6, 32.9887], np.float64),
+                str=["Hello", "World", "Foo", "Bar"],
+                enum=[ExampleEnum.A, ExampleEnum.B, ExampleEnum.A, ExampleEnum.C],
+            ),
+        ),
+        EverythingSignal("ndarray", np.ndarray, np.array(([1, 2, 3], [4, 5, 6]))),
+    ]
```

Comment:
Although this saves typing, it's actually pretty useful to see how you would make the Tango Device and it's associated ophyd device (and the same for EPICS and sim). Rather than try to be clever here I think I'd prefer to just see the test data written out longhand, with this file not being imported from Tango...